mlir/include/mlir/Dialect/Rock/utility/loweringUtils.h

```diff
@@ -22,6 +22,76 @@ namespace rock {
 struct ConvolutionDims;
 struct GemmSize;
 
+// This is a helper data structure
+// to server as a return type for
+// view creating utilities.
+struct GPUViews {
+  // This is a [gridIdx0, ... , gridIdxN, tid, iter] to K x D view
+  // This is useful to define a global tile once loaded into regs.
+  ArrayAttr gridwiseView;
+  // This is a [tid, iter] to flat K x D view
+  // Useful to define a blockwise tile once loaded into regs.
+  ArrayAttr blockwiseView;
+  // This is a [iter] to flat K x D view
+  // Useful to define a threadwise tile once loaded into regs.
+  ArrayAttr threadwiseView;
+};
+
+// This is helper data structure to allow
+// lowering define lower dimensions (K or D) re-shuffling
+// into thread_id or iter into reg buffers in threads.
+// It is particularly useful to arrange data
+// in registers before they are loaded/written
+// to some memory (global or lds).
+struct LowerDimPartInfo {
```

Comment:
I'm not liking these.

It doesn't seem like you can quite accomplish what you're going for by just returning a `rock::TransformAttr` and some result sizes, but ... maybe? 

Especially if the transform map builders grew an `insertTransform(TransformAttr, ArrayRef<int64_t>)`.